editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -1936,6 +1940,122 @@ bool VisualShaderEditor::_is_available(int p_mode) {
 	return (p_mode == -1 || (p_mode & current_mode) != 0);
 }
 
+bool VisualShaderEditor::_update_preview_parameter_tree() {
+	bool found = false;
+	bool use_filter = !param_filter_name.is_empty();
+
+	parameters->clear();
+	TreeItem *root = parameters->create_item();
+
+	for (const KeyValue<String, PropertyInfo> &prop : parameter_props) {
+		String param_name = prop.value.name;
+		if (use_filter && !param_name.containsn(param_filter_name)) {
+			continue;
+		}
+
+		TreeItem *item = parameters->create_item(root);
+		item->set_text(0, param_name);
+		item->set_meta("id", param_name);
+
+		if (param_name == selected_param_id) {
+			parameters->set_selected(item);
+			found = true;
+		}
+
+		switch (prop.value.type) {
+			case Variant::BOOL: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("bool")));
+			} break;
+			case Variant::INT: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("int")));
+			} break;
+			case Variant::FLOAT: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("float")));
+			} break;
+			case Variant::VECTOR2: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("Vector2")));
+			} break;
+			case Variant::VECTOR3: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("Vector3")));
+			} break;
+			case Variant::VECTOR4: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("Vector4")));
+			} break;
+			case Variant::COLOR: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("Color")));
+			} break;
+			case Variant::TRANSFORM3D: {
+				item->set_icon(0, get_editor_theme_icon(SNAME("Transform3D")));
+			} break;
```

Comment:
You can do
item->set_icon(0, get_editor_theme_icon(Variant::get_type_name(prop.value.type)));